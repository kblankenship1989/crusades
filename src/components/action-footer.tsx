import React from 'react';
import { Icon, Button } from 'react-native-elements';
import { ActionFooterProps } from '../types/components/props';
import { View } from 'react-native';

export const ActionFooter: React.FC<ActionFooterProps> = ({ onAdd, onSave }) => {
    return (
        <View>
            {onAdd && <Button
                onPress={onAdd}
                icon={<Icon
                    name={'plus'}
                    type={'font-awesome'}
                    size={18}
                />}
                title={'ADD'}
            />}
            {onSave && <Button
                onPress={onAdd}
                icon={<Icon
                    name={'save'}
                    type={'font-awesome'}
                    size={18}
                />}
                title={'SAVE'}
            />}
        </View>
    );
};
