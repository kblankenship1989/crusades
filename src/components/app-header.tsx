import React from 'react';
import {Icon, IconButton} from 'native-base';
import {AppBar} from 'native-base/src/components/composites/AppBar';

export type Action = {
    icon: string,
    onPress: () => void,
    isDisabled: boolean
}

type AppHeaderProps = {
    title: string,
    action?: Action
}

export const AppHeader : React.FC<AppHeaderProps> = ({
    title,
    action
}) => {

    return (
        <AppBar>
            <AppBar.Left/>
            <AppBar.Content>{title}</AppBar.Content>
            {action &&
                <AppBar.Right>
                    <IconButton
                        onPress={action.onPress}
                        disabled={action.isDisabled}
                        icon={
                            <Icon
                                type={'MaterialCommunityIcons'}
                                name={action.icon}
                            />
                        }
                    />
                </AppBar.Right>
            }
        </AppBar>
    );
};
