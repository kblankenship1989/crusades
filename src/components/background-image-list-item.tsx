import React from 'react';
import {TouchableOpacity, Text, StyleSheet, Alert} from 'react-native';
import {BackgroundImageContainer} from '../containers/background-image-container';
import {BattlefieldRoles, Factions} from '../types/enums';

type BackgroundImageListItemProps = {
    onDelete: (index: number) => void,
    onPress: (index: number) => void,
    index: number,
    title: string,
    imageKey: Factions | BattlefieldRoles
}

const styles = StyleSheet.create({
    title: {
        fontWeight: 'bold',
        opacity: 1,
        fontSize: 24
    },
    listItemContainer: {
        flex: 1
    },
    imageContainer: {
        flex: 1
    }
});

export const BackgroundImageListItem = ({
    onPress,
    onDelete,
    index,
    title,
    imageKey
} : BackgroundImageListItemProps) : JSX.Element => {

    const createConfirmDeleteAlert = () => {
        Alert.alert(
            'Confirm Delete',
            'Are you sure you want to delete this item? There is no undo',
            [
                {
                    text: 'Cancel',
                    style: 'cancel'
                },
                {
                    text: 'Delete',
                    onPress: () => onDelete(index),
                    style: 'destructive'
                }
            ],
            {cancelable: true}
        );

    };

    return (
        <TouchableOpacity
            onPress={() =>  onPress(index)}
            onLongPress={createConfirmDeleteAlert}
            style={styles.listItemContainer}
        >
            <BackgroundImageContainer
                containerStyle={styles.imageContainer}
                imageKey={imageKey}
            >
                <Text style={styles.title}>
                    {title}
                </Text>
            </BackgroundImageContainer>
        </TouchableOpacity>
    );
};
