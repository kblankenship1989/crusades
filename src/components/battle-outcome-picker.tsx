import React from 'react';
import {enumKeys, isEnumKey} from '../helpers/enum-helpers';
import {BattleOutcomes} from '../enums';

import {Picker} from './picker';

export type BattleOutcomePickerProps = {
    selectedBattleOutcome?: BattleOutcomes,
    onChange: (faction: BattleOutcomes) => void
}

type Item = {
    key: keyof typeof BattleOutcomes,
    value: BattleOutcomes
}

const items : Item[] = enumKeys(BattleOutcomes).map((key) => ({
    key,
    value: BattleOutcomes[key]
}));

export const BattleOutcomePicker : React.FC<BattleOutcomePickerProps> = ({selectedBattleOutcome, onChange}) => {
    const onValueChange = (item : React.ReactText) : void => {
        if (isEnumKey(BattleOutcomes)(item)) {
            onChange(item);
        }
    };

    return (
        <Picker
            selectedValue={selectedBattleOutcome}
            placeholder={'Select Outcome'}
            items={items}
            onChange={onValueChange}
        />
    );
};

