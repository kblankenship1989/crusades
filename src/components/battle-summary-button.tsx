import {useNavigation} from '@react-navigation/native';
import {StackNavigationProp} from '@react-navigation/stack';
import React from 'react';
import {Button, View} from 'react-native';
import {Text} from 'react-native-elements';
import {BattleOutcomes} from '../types/enums';

export type BattleSummaryButtonProps = {
    battleTally: BattleResults[]
}

type WinLoseDraw = 'WIN' | 'LOSE' | 'DRAW';

const WinLoseDrawMap : Record<BattleOutcomes, WinLoseDraw> = {
    [BattleOutcomes.TABLED]: 'LOSE',
    [BattleOutcomes.MAJOR_LOSS]: 'LOSE',
    [BattleOutcomes.MINOR_LOSS]: 'LOSE',
    [BattleOutcomes.DRAW]: 'DRAW',
    [BattleOutcomes.MINOR_VICTORY]: 'WIN',
    [BattleOutcomes.MAJOR_VICTORY]: 'WIN',
    [BattleOutcomes.ANNIHALATION]: 'WIN'
};

export const BattleSummaryButton : React.FC<BattleSummaryButtonProps> = ({battleTally}) : JSX.Element => {
    const navigation : StackNavigationProp<RootParamList, 'OrderOfBattleSummary'> = useNavigation();

    const navigateToBattleSummary = () : void => {
        navigation.push('BattleSummary', {
            battleResults: battleTally,
        });
    };

    const getWinLoseDrawCounts = () : string => {
        const wldCounts : Record<WinLoseDraw, number> = battleTally.reduce((counts, battleResult) => {
            counts[WinLoseDrawMap[battleResult.result]]++;

            return counts;
        }, {
            WIN: 0,
            LOSE: 0,
            DRAW: 0
        });

        return `${wldCounts.WIN} / ${wldCounts.LOSE} / ${wldCounts.DRAW}`;
    };

    return (
        <View>
            <Text h3>Battle Summary (W/L/D)</Text>
            <Button
                title={getWinLoseDrawCounts()}
                onPress={navigateToBattleSummary}
            />
        </View>
    );
};
