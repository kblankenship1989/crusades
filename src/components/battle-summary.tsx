import React from 'react';
import {Card} from 'react-native-elements';
import {Text, Button} from 'react-native';
import {Picker} from '@react-native-community/picker';
import {getColorScheme} from '../helpers/getColorScheme';
import {appStyles} from '../../styles';
import {BattleOutcomes, battleOutcomes} from '../types/consts';

export type BattleSummaryProps = {
    battleTallies: BattleOutcomes[],
    addBattleTally: (battleOutcome : BattleOutcomes) => void
}

const colorScheme = getColorScheme();
const styles = appStyles(colorScheme);

export const BattleSummary = ({battleTallies, addBattleTally} : BattleSummaryProps) : JSX.Element => {
    const [currentSelection, setCurrentSelection] = React.useState<BattleOutcomes | undefined>(undefined);

    const updateCurrentSelection = (itemValue: string | number, itemIndex : number) : void => {
        setCurrentSelection(battleOutcomes[itemIndex]);
    };

    const addNewBattleTally = () : void => {
        if (currentSelection !== undefined) {
            addBattleTally(currentSelection);
        }

        setCurrentSelection(undefined);
    };

    enum WinLoseDraw {
        Win = 'Win',
        Lose = 'Lose',
        Draw = 'Draw'
    }

    interface WinLoseDrawSummary {
        [WinLoseDraw.Win]: number,
        [WinLoseDraw.Lose]: number,
        [WinLoseDraw.Draw]: number
    }

    const addTallyToSummary = (summary: WinLoseDrawSummary, key: keyof WinLoseDrawSummary) : WinLoseDrawSummary => {
        return {
            ...summary,
            [key]: summary[key] + 1
        };
    };

    const winLoseDrawMap : Record<BattleOutcomes, WinLoseDraw> = {
        [battleOutcomes[0]]: WinLoseDraw.Lose,
        [battleOutcomes[1]]: WinLoseDraw.Lose,
        [battleOutcomes[2]]: WinLoseDraw.Lose,
        [battleOutcomes[3]]: WinLoseDraw.Draw,
        [battleOutcomes[4]]: WinLoseDraw.Win,
        [battleOutcomes[5]]: WinLoseDraw.Win,
        [battleOutcomes[6]]: WinLoseDraw.Win
    };

    const wldSummary : WinLoseDrawSummary = battleTallies.reduce((summary : WinLoseDrawSummary, battleTally : BattleOutcomes) : WinLoseDrawSummary => {
        const key : WinLoseDraw = winLoseDrawMap[battleTally];

        return addTallyToSummary(summary, key);

    }, {[WinLoseDraw.Win]: 0, [WinLoseDraw.Lose]: 0, [WinLoseDraw.Draw]: 0});

    return (
        <Card
            title={'Battle Summary'}
        >
            <Text>W / L / D</Text>
            <Text>{`${wldSummary[WinLoseDraw.Win]} / ${wldSummary[WinLoseDraw.Lose]} / ${wldSummary[WinLoseDraw.Draw]}`}</Text>
            <Picker
                accessibilityLabel={'Faction'}
                testID={'faction-picker'}
                style={styles.picker}
                selectedValue={currentSelection}
                onValueChange={updateCurrentSelection}
            >
                {battleOutcomes.map((battleOutcome) => (
                    <Picker.Item
                        label={battleOutcome}
                        value={battleOutcome}
                        key={battleOutcome}
                    />
                ))}
            </Picker>
            <Button
                onPress={addNewBattleTally}
                title={'Add Battle'}
            />
        </Card>
    );
};
