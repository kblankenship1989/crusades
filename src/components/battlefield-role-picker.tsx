import React from 'react';
import {enumKeys, isEnumKey} from '../helpers/enum-helpers';
import {BattlefieldRoles} from '../enums';

import {Picker} from './picker';
import {View, Separator, Text} from 'native-base';

export type BattlefieldRolePickerProps = {
    selectedRole?: BattlefieldRoles,
    onChange: (role: BattlefieldRoles) => void,
    placeholder?: string,
    title: string
}

type Item = {
    key: keyof typeof BattlefieldRoles,
    value: BattlefieldRoles
}

const items : Item[] = enumKeys(BattlefieldRoles).map((key) => ({
    key,
    value: BattlefieldRoles[key]
}));

export const BattlefieldRolePicker : React.FC<BattlefieldRolePickerProps> = ({selectedRole, onChange, placeholder, title}) => {
    const onValueChange = (item : React.ReactText) : void => {
        if (isEnumKey(BattlefieldRoles)(item)) {
            onChange(item);
        }
    };

    return (
        <View>
            <Separator>
                <Text>{title}</Text>
            </Separator>
            <Picker
                selectedValue={selectedRole}
                items={items}
                onChange={onValueChange}
                placeholder={placeholder || 'Select Battlefield Role'}
            />
        </View>
    );
};

