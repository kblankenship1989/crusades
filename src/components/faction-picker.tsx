import React from 'react';
import {Picker} from '@react-native-community/picker';
import {getColorScheme} from '../helpers/getColorScheme';
import {appStyles} from '../../styles';
import {View} from 'react-native';
import {Icon} from 'react-native-elements';
import {factions} from '../types/consts';

export type FactionPickerProps = {
    selectedFaction: string,
    onValueChange: (itemValue: string | number, itemIndex: number) => void
}
const colorScheme = getColorScheme();
const styles = appStyles(colorScheme);

export const FactionPicker = (props : FactionPickerProps) : JSX.Element => (
    <View style={styles.pickerRow}>
        <Icon
            size={18}
            name={'sword-cross'}
            type={'material-community'}
        />
        <Picker
            accessibilityLabel={'Faction'}
            testID={'faction-picker'}
            style={styles.picker}
            selectedValue={props.selectedFaction}
            onValueChange={props.onValueChange}
        >
            {factions.map((faction) => (
                <Picker.Item
                    label={faction}
                    value={faction}
                    key={faction}
                />
            ))}
        </Picker>
    </View>
);
