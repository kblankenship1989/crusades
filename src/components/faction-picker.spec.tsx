import React from 'react';
import {FactionPicker} from './faction-picker';
import {factions} from '../types/consts';
import {render} from '@testing-library/react-native';
import {FactionPickerProps} from '../types/components/props';

describe('Given the Faction Picker Component', () => {
    it('should render correctly', () => {
        const testProps : FactionPickerProps = {
            selectedFaction: factions[3],
            onChange: jest.fn()
        };
        const component = (<FactionPicker
            {...testProps}
        />);

        expect(render(component).toJSON()).toMatchSnapshot();
    });
});
