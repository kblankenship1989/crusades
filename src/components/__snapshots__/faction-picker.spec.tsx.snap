// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Faction Picker Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "zIndex": 5000,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderColor": "#dfdfdf",
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {},
            Object {
              "flex": 1,
              "opacity": 1,
            },
            Object {
              "marginLeft": 5,
              "marginRight": 0,
            },
            Object {
              "marginLeft": 5,
            },
            Object {},
            Object {},
            false,
          ]
        }
      >
        Aeldari
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomLeftRadius": 0,
              "borderTopLeftRadius": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderColor": "#dfdfdf",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "borderWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          },
          Object {
            "alignItems": "center",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "justifyContent": "center",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "#fafafa",
          },
          Object {
            "borderWidth": 0,
            "display": "none",
            "position": "relative",
          },
          Object {
            "maxHeight": 150,
            "top": undefined,
            "zIndex": 5000,
          },
        ]
      }
    >
      <RCTScrollView
        nestedScrollEnabled={true}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Unaligned
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Imperium
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Chaos
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Aeldari
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Tyranids
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Orks
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  Necrons
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "opacity": 1,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {},
                      Object {},
                      undefined,
                      false,
                      Object {},
                    ]
                  }
                >
                  T'au Empire
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
